main/exercise/ptest_agree_scale.class.php

```diff
@@ -0,0 +1,346 @@
+                $categoriesOptions
+            );
+
+            $form->addHtml('</tr>');
+        }
+
+        $form->addHtml('</tbody>');
+        $form->addHtml('</table>');
+
+        global $text;
+        $buttonGroup = [];
+
+        if ($obj_ex->edit_exercise_in_lp == true ||
+            (empty($this->exerciseList) && empty($obj_ex->id))
+        ) {
+            //setting the save button here and not in the question class.php
+            $buttonGroup[] = $form->addButtonDelete(get_lang('LessAnswer'), 'lessAnswers', true);
+            $buttonGroup[] = $form->addButtonCreate(get_lang('PlusAnswer'), 'moreAnswers', true);
+            $buttonGroup[] = $form->addButton(
+                'submitQuestion',
+                $text,
+                'check',
+                'primary',
+                'default',
+                null,
+                ['id' => 'submit-question'],
+                true
+            );
+            $form->addGroup($buttonGroup);
+        }
+
+        if (!empty($this->id)) {
+            $form->setDefaults($defaults);
+        } else {
+            if ($this->isContent == 1) {
+                // Default sample content.
+                $form->setDefaults($defaults);
+            }
+        }
+        $form->setConstants(['nb_answers' => $nb_answers]);
+    }
+
+    public function setDirectOptions($i, FormValidator $form, $renderer, $select_lp_id, $select_question)
+    {
+        $editor_config = [
+            'ToolbarSet' => 'TestProposedAnswer',
+            'Width' => '100%',
+            'Height' => '125',
+        ];
+
+        $form->addHtmlEditor(
+            'comment['.$i.']',
+            null,
+            null,
+            false,
+            $editor_config
+        );
+        // Direct feedback
+        //Adding extra feedback fields
+        $group = [];
+        $group['try'.$i] = $form->createElement(
+            'checkbox',
+            'try'.$i,
+            null,
+            get_lang('TryAgain')
+        );
+        $group['lp'.$i] = $form->createElement(
+            'select',
+            'lp'.$i,
+            get_lang('SeeTheory').': ',
+            $select_lp_id
+        );
+        $group['destination'.$i] = $form->createElement(
+            'select',
+            'destination'.$i,
+            get_lang('GoToQuestion').': ',
+            $select_question
+        );
+        $group['url'.$i] = $form->createElement(
+            'text',
+            'url'.$i,
+            get_lang('Other').': ',
+            [
+                'class' => 'col-md-2',
+                'placeholder' => get_lang('Other'),
+            ]
+        );
+        $form->addGroup($group, 'scenario');
+
+        $renderer->setElementTemplate(
+            '<td><!-- BEGIN error --><span class="form_error">{error}</span><!-- END error --><br/>{element}',
+            'scenario'
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function processAnswersCreation($form, $exercise)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The variable $nb_answers is not named in camelCase.](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5018478)